ouroboros-consensus/test-consensus/Test/Consensus/Ledger/Byron.hs

```diff
@@ -208,22 +215,56 @@ prop_byronHashInfo_hashSize h =
   Golden tests
 -------------------------------------------------------------------------------}
 
+-- | Note that we must use the same value for the 'SecurityParam' as for the
+-- 'CS.WindowSize', because 'decodeByronChainState' only takes the
+-- 'SecurityParam' and uses it as the basis for the 'CS.WindowSize'.
+secParam :: SecurityParam
+secParam = SecurityParam 2
+
+windowSize :: CS.WindowSize
+windowSize = CS.WindowSize 2
+
+exampleChainStateWithoutEBB, exampleChainStateWithEBB :: ChainState (BlockProtocol ByronBlock)
+(exampleChainStateWithoutEBB, exampleChainStateWithEBB) =
+    (withoutEBB, withEBB)
+  where
+    signers = map (`CS.PBftSigner` CC.exampleKeyHash) [1..4]
+    hashBytes = mkHeaderHashBytesForTestingOnly (Lazy8.pack "test_golden_ChainState6")
+
+    withoutEBB = CS.fromList
+      secParam
+      windowSize
+      (At 2, Seq.fromList signers, CS.EbbMap mempty)
+
+    withEBB = CS.appendEBB secParam windowSize 6 hashBytes withoutEBB
+
 test_golden_ChainState :: TestTree
 test_golden_ChainState = goldenTestCBOR
     "ChainState"
     encodeByronChainState
-    exampleChainState
-    "test-consensus/golden/cbor/byron/ChainState"
-  where
-    exampleChainState = CS.appendEBB secParam windowSize 6 $
-      CS.fromList
-        secParam
-        windowSize
-        (At 3, Seq.fromList signers, CS.EbbMap mempty)
-
-    secParam = SecurityParam 2
-    windowSize = CS.WindowSize 3
-    signers = map (`CS.PBftSigner` CC.exampleKeyHash) [1..5]
+    exampleChainStateWithEBB
+    "test-consensus/golden/cbor/byron/ChainState2"
+
+test_golden_ChainState_backwardsCompat_version0 :: TestTree
+test_golden_ChainState_backwardsCompat_version0 =
+    testCase "ChainState version 0" $ goldenTestCBORBackwardsCompat
+      (decodeByronChainState secParam)
+      exampleChainStateWithoutEBB
+      "test-consensus/golden/cbor/byron/ChainState0"
+
+test_golden_ChainState_backwardsCompat_version1 :: TestTree
+test_golden_ChainState_backwardsCompat_version1 =
+    testCase "ChainState version 1" $ goldenTestCBORBackwardsCompat
+      (decodeByronChainState secParam)
+      exampleChainStateWithoutEBB
+      "test-consensus/golden/cbor/byron/ChainState1"
```

Comment:
@mrBliss This tests the ability to deserialize a V1 blob that arose with an empty `EbbMap`. Should we also test a V1 blob that ha a non-empty `EbbMap`? It'll be a bit tedius to generate one. And if I understand correctly we've probably never actually generated one in practice, right? Since we have no recent EBBs.

Reply:
Meh, it would be nice, but I don't care much about it.